cpp/examples/arrow/CMakeLists.txt

```diff
@@ -84,6 +84,24 @@ if(ARROW_FLIGHT)
                     EXTRA_SOURCES
                     "${CMAKE_CURRENT_BINARY_DIR}/helloworld.pb.cc"
                     "${CMAKE_CURRENT_BINARY_DIR}/helloworld.grpc.pb.cc")
+
+  if(ARROW_FLIGHT_SQL)
+    if(ARROW_GRPC_USE_SHARED)
+      set(FLIGHT_SQL_EXAMPLES_LINK_LIBS arrow_flight_sql_shared)
+    else()
+      set(FLIGHT_SQL_EXAMPLES_LINK_LIBS arrow_flight_sql_static)
+    endif()
+
+    add_arrow_example(flight_sql_example
```

Comment:
Looks we prefer `-` for executable names, `_` for source names.
Change it to `flight-sql-example`?

Reply:
Hmm, seems originally file names and executable names used hyphens, then file names were switched to underscores. We can file an issue to do the conversion, since none of the examples or other executables follow this convention right now.

Reply:
I filed ARROW-15625.